### ISSUE TITLE
require-local-test-context-for-concurrent-snapshots should detect Vitest options object concurrency

### ISSUE THREAD

**Issue by nami8824** [vitest-dev/eslint-plugin-vitest#898]
## Summary

`require-local-test-context-for-concurrent-snapshots` currently misses concurrent tests declared via Vitest's options object:

```ts
test('snapshot test', { concurrent: true }, () => {
  expect(1).toMatchSnapshot()
})
```

This form is currently not reported.

Since this test is concurrent, snapshot assertions should use the local Test Context expect instead of the global expect.

## Reproduction

Add the following case to the invalid section of `tests/require-local-test-context-for-concurrent-snapshots.test.ts.`

```ts
{
  code: 'it("snapshot test", { concurrent: true }, () => { expect(1).toMatchSnapshot() })',
  errors: [{ messageId: 'requireLocalTestContext' }],
},
```

Then run:

```
pnpm test tests/require-local-test-context-for-concurrent-snapshots.test.ts
```
Currently, this case fails because the rule only checks the `.concurrent` chain form and does not recognize `{ concurrent: true }`.
